dynamic/json_test.go

```diff
@@ -87,8 +88,43 @@ func TestMarshalJSONIndent(t *testing.T) {
 	testutil.Eq(t, "{\n\t\"foo\": [\n\t\t\"VALUE1\"\n\t],\n\t\"bar\": \"bedazzle\"\n}", string(jsIndent))
 }
 
+func TestUnmarshalJSON(t *testing.T) {
```

Comment:
I don't think this tests anything new. There isn't anything in `TestRequest` that isn't already covered by existing cases (`TestUnaryFieldsJSON`, `TestRepeatedFieldsJSON`, `TestMapKeyFieldsJSON`, and `TestMapValueFieldsJSON`).